File: monRDV-boot-rest/src/main/java/monRDV/model/CreneauDisponible.java
package monRDV.model;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.persistence.Version;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@Table(name = "creneauDisponible")
public class CreneauDisponible {
	@Id
	@GeneratedValue
	@JsonView(Views.ViewCommon.class)
	private Long id;
	@Version
	@JsonView(Views.ViewCommon.class)
	private int version;
	@Temporal(TemporalType.TIMESTAMP)
	@Column(name = "date_debut")
	@JsonView(Views.ViewCommon.class)
	private Date debut;
	@Temporal(TemporalType.TIMESTAMP)
	@Column(name = "date_fin")
	@JsonView(Views.ViewCommon.class)
	private Date fin;
	@ManyToOne
	@JoinColumn(name = "lieu_id")
	@JsonView(Views.ViewCommon.class)
	private Lieu lieu;
	
	@ManyToOne
	@JsonView(Views.ViewPraticien.class)
	private Praticien praticien;
	
	@ManyToOne
	@JsonView(Views.ViewCreneauDisponible.class)
	private RendezVous rendezVous;

	public CreneauDisponible() {
		super();
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public int getVersion() {
		return version;
	}

	public void setVersion(int version) {
		this.version = version;
	}

	public Date getDebut() {
		return debut;
	}

	public void setDebut(Date debut) {
		this.debut = debut;
	}

	public Date getFin() {
		return fin;
	}

	public void setFin(Date fin) {
		this.fin = fin;
	}

	public Lieu getLieu() {
		return lieu;
	}

	public void setLieu(Lieu lieu) {
		this.lieu = lieu;
	}

	public Praticien getPraticien() {
		return praticien;
	}

	public void setPraticien(Praticien praticien) {
		this.praticien = praticien;
	}

	public RendezVous getRendezVous() {
		return rendezVous;
	}

	public void setRendezVous(RendezVous rendezVous) {
		this.rendezVous = rendezVous;
	}



}

File: monRDV-angular/src/app/model/creneauDisponible.ts
import {RendezVous} from './rendezVous';
import {Lieu} from './lieu';
import {Praticien} from './praticien';

export class CreneauDisponible {
  private _id: number;
  private _version: number;
  private _debut: Date;
  private _lieu: Lieu;
  private _praticien: Praticien;
  private _rendezVous: RendezVous;

  constructor(id?: number, version?: number, debut?: Date, lieu?: Lieu, praticien?: Praticien, rendezVous?: RendezVous) {
    this._id = id;
    this._version = version;
    this._debut = debut;
    this._lieu = lieu;
    this._praticien = praticien;
    this._rendezVous = rendezVous;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get version(): number {
    return this._version;
  }

  set version(value: number) {
    this._version = value;
  }

  get debut(): Date {
    return this._debut;
  }

  set debut(value: Date) {
    this._debut = value;
  }

  get lieu(): Lieu {
    return this._lieu;
  }

  set lieu(value: Lieu) {
    this._lieu = value;
  }

  get praticien(): Praticien {
    return this._praticien;
  }

  set praticien(value: Praticien) {
    this._praticien = value;
  }

  get rendezVous(): RendezVous {
    return this._rendezVous;
  }

  set rendezVous(value: RendezVous) {
    this._rendezVous = value;
  }
}
